### PR TITLE
Stage diag and field tables.

### PR DESCRIPTION
**Pull request by aerorahul**
**Description**

This PR:
- breaks the `diag_table_da` and `diag_table` into `atm` and `ocn` counterparts so they can be assembled based on the configuration of the model
- adds a `diag_table.header` that contains the first two lines of the diag table header.
- adds methods to stage the tables, as well as parse the information into it
- adds a `ufs_model` dict container in `UFS.py` to store model specific information during the `initialize` phase of the task.  More attributes will be added to this dict as the task matures.

**Type of change**

- [X] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

Run on hera and the run directory was inspected and found to contain the fix files as well as the tables, parsed and ready for use.  E.g. `/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNDIRS/atmcyc3d/fcst.161253/`

  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: parm/parm_fv3diag/diag_table.ocn [hunk 1]
```diff
@@ -0,0 +1,71 @@
+"ocn%4yr%2mo%2dy%2hr",    6,  "hours",  1,  "hours",  "time",  6,  "hours",  "1901 1 1 0 0 0"
```
Comment (JessicaMeixner-NOAA) [inline parm/parm_fv3diag/diag_table.ocn:1]: When combining the two atm & ocn diag file segments, do these two lines need to be at the top of the file along with the 
```
"fv3_history",    0,  "hours",  1,  "hours",  "time"
"fv3_history2d",  0,  "hours",  1,  "hours",  "time"
```
section?

Comment (aerorahul) [reply parm/parm_fv3diag/diag_table.ocn:1]: No. 

See the File Description and Field Description slides from https://data1.gfdl.noaa.gov/summer-school/Lectures/July16/03_Seth1_DiagManager.pdf
The only requirement is that the file be described before using it for a field.

Comment (JessicaMeixner-NOAA) [reply parm/parm_fv3diag/diag_table.ocn:1]: Thanks!

**Review by HenryRWinterbottom — APPROVED** [review]
These look good to me.

File: ush/python/pygfs/ufswm/gfs.py [hunk 2]
```diff
@@ -14,21 +15,156 @@ class GFS(UFS):
...
+        """
+        Description
+        -----------
+        This method extracts configuration directly from `self._config` and maps it to `self.ufs_model`
+        TODO: This method could be broken up into smaller methods for each component, but maintain this as the entry point
+
+        Parameters
+        ----------
+        None
+        Returns
```
Comment (WalterKolczynski-NOAA) [inline ush/python/pygfs/ufswm/gfs.py:40]:
```Suggestion
        None

        Returns
```

File: ush/python/pygfs/ufswm/gfs.py [hunk 3]
```diff
@@ -14,21 +15,156 @@ class GFS(UFS):
...
+        logger.debug(f"UFS-weather-model configuration:\n{pformat(self.ufs_model)}")
+
+    @logit(logger)
+    def _get_fix_info(self) -> None:
+        self.ufs_model.fix = AttrDict()
+        self.ufs_model.fix.FIX_aer = os.path.join(self.ufs_model.HOMEufs, 'fix', 'aer')
+        self.ufs_model.fix.FIX_am = os.path.join(self.ufs_model.HOMEufs, 'fix', 'am')
+        self.ufs_model.fix.FIX_lut = os.path.join(self.ufs_model.HOMEufs, 'fix', 'lut')
+        self.ufs_model.fix.FIX_orog = os.path.join(self.ufs_model.HOMEufs, 'fix', 'orog')
+        self.ufs_model.fix.FIX_ugwd = os.path.join(self.ufs_model.HOMEufs, 'fix', 'ugwd')
```
Comment (WalterKolczynski-NOAA) [inline ush/python/pygfs/ufswm/gfs.py:64]: Why the extra `FIX_` in the attribute names?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
A couple very small things. Since this was just merged, they can be fixed in a future PR.

File: ush/python/pygfs/ufswm/gfs.py (hunk 3 above)
Comment (aerorahul) [reply ush/python/pygfs/ufswm/gfs.py:64]: so I can use `FIX_` directly in the template.  I agree there is redundancy here, but in the yaml, the template is clear.
The template will be `$(FIX_ugwd)`.  If I remove the `FIX_`, then the template will be `$(ugwd)`.  The latter is less intuitive, infact confusing.

**Comment by aerorahul** [comment]
Note that this PR is into `dev/fcstsprint` and not `develop`.  
Before we merge to `develop`, there will be a wider and more thorough review from a variety of developers.

**Pull request merged**